src/server/worker/api_server.go

```diff
@@ -676,6 +677,9 @@ func (a *APIServer) uploadOutput(pachClient *client.APIClient, dir string, tag s
 		if err != nil {
 			return err
 		}
+		if !utf8.ValidString(filePath) {
+			return fmt.Errorf("Filepath is not valid utf-8: %s", filePath)
```

Comment:
Very minor nit, there's a general standard in Go that error messages shouldn't start with capital letters because they're normally prefixed in other messages. I.e. this will be emitted to logs as "Error uploading output: Filepath is not..." which reads a little bit weirdly.